### PR TITLE
Remove create-react-context and bump React peer dependency

### LINKED ISSUE #213: Generics support for Component
Hi! 

I'm the maintainer of https://github.com/dotansimha/graphql-code-generator . 
[One of our community members suggested we should add a plugin](https://github.com/dotansimha/graphql-code-generator/issues/1575) that generates ready-to-use type-safe Components and hooks using your library. 
It sounds like a cool idea, but we currently not able to implement it because your implementation does not allow to override the type of `data` (https://github.com/FormidableLabs/urql/blob/master/src/components/Query.tsx#L16) or `variables`.

In the hooks implementation, we can pass the type using generics and override the `data` and `variables` types, which is great, because we can generate ready-to-use wrappers for it. 

It would be great to have support for generics, and then we'll implement the plugin for the codegen :)

### PR DESCRIPTION
**Pull request by kitten**
It's probably been a mistake to include `create-react-context` and support non-hook implementations in practice. This will basically get rid of most of the maintenance burden we have right now, since we won't have to maintain two APIs for each hook/component.

Fix #213

### CODE REVIEW

File: src/components/Mutation.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export interface MutationState<T, V> extends UseMutationState<T> {
+  executeMutation: (variables?: V) => Promise<OperationResult<T>>;
+}
+
+export function Mutation<T = any, V = any>({
+  children,
+  query,
+}: MutationProps<T, V>): ReactNode {
+  const [state, executeMutation] = useMutation<T, V>(query);
```
Comment (JoviDeCroock) [inline src/components/Mutation.ts]: This change requires a peerDep of React 16.8

File: src/components/Query.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+export interface QueryState<T> extends UseQueryState<T> {
+  executeQuery: (opts?: Partial<OperationContext>) => void;
+}
+
+export function Query<T = any, V = any>({
+  children,
+  ...args
+}: QueryProps<T, V>): ReactNode {
+  const [state, executeQuery] = useQuery<T, V>(args);
```
Comment (JoviDeCroock) [inline src/components/Query.ts]: Same here, I don't think we can set peerdep to 16.6.1 is we're using hooks in our render methods

Comment (kitten) [reply src/components/Query.ts]: Whoops

File: src/components/Subscription.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export interface SubscriptionProps<T, V, R = T> extends UseSubscriptionArgs<V> {
+  handler?: SubscriptionHandler<T, R>;
+  children: (arg: UseSubscriptionState<R>) => ReactNode;
+}
+
+export function Subscription<T = any, R = T, V = any>({
+  children,
+  handler,
+  ...args
+}: SubscriptionProps<T, V, R>): ReactNode {
```
Comment (parkerziegler) [inline src/components/Subscription.ts]: Super nit – it looks like `SubscriptionProps` receives its generics in the order `<T, V, R>`, but `Subscription`'s generics (and `useSubscription`) are in the order `<T, R, V>`. I'd keep the ordering consistent if possible here just to keep these signatures easy to read.

Comment (kitten) [reply src/components/Subscription.ts]: Good catch! 👍

**Pull request merged**